packages/web3wallet/src/utils/notifications.ts

```diff
@@ -0,0 +1,24 @@
+import { Core } from "@walletconnect/core";
+import { Web3WalletTypes } from "../types";
+import { SessionStore } from "@walletconnect/sign-client";
+
+export const Notifications: Web3WalletTypes.INotifications = {
+  decryptMessage: async (params) => {
+    const core = new Core({
```

Comment:
Bit concerned regarding memory and creating a new Core reference on every `decryptMessage` call. Is there any way to avoid this and init the Core once? Guessing this may have to be stateless right.

Reply:
I don't think we have a way around this because our modules are not stateless/pure - crypto depends on `core` and having a store AFAIK

Reply:
Good point 💯  I will thinker a solution as we do need to reinitialize to update the data